tests/components/wemo/test_entity.py

```diff
@@ -0,0 +1,104 @@
+"""Tests exception cases in entity.py."""
```

Comment:
There should be platform tests to cover the common code so we shouldn't add tests in a new module. If more tests are needed they should be added in the platform test modules.

Reply:
It would be good to add any required tests before doing this refactor.

Reply:
@MartinHjelmare, Adding the tests to each platform module sounds good to me. Since most of the platforms share the same code, the tests will also be the same. I'd like to avoid copy/pasting the tests too, if that's okay with you.

Here's my suggestion:
- Create a new PR with test_entity.py renamed to entity_test_methods.py.
- In each platform test file where the test make sense, import the test from entity_test_methods.py (`from .entity_test_methods import <test_method> [<test_method>...]` and clearly document what this is doing (i.e. Importing tests that are common between platform entities. These imported methods names begin with 'test_' and thus will be run as part of this test module.)
- Remove test_entity.py from this PR.

I was trying to make these tests shared in #38959, but shouldn't have done so with classes. I just want to make sure that newly suggested approach makes sense before I move forward with it. If there's a more appropriate way, or some examples of other tests that do similar things, please let me know too. Also, if I'm off base with this PR, and convention is to have common code copied into each platform, please let me know too. :)

Reply:
Each platform should test its own entity type and services. It's ok to use common fixtures and test helpers but the tests should be defined in each platform.

Reply:
I've moved the tests to #44136. Let's wait for that PR before reviewing this PR.

> Each platform should test its own entity type and services. It's ok to use common fixtures and test helpers but the tests should be defined in each platform.

Thanks for this! A bug was uncovered while working on #44136 as a result of your advice.

Reply:
Will the coverage check be resolved here if we rebase now on latest dev?

Reply:
The coverage check did pass before, when the tests were included in this PR. The `async_will_remove_from_hass` method is still untested because the component currently doesn't support unloading (hopefully coming in a future PR). Besides that one method, everything else from this PR is covered by the tests in #44136.

Would you like me to rebase this PR to the latest dev branch? I think a force push will re-trigger all the checks.

Reply:
It's ok to rebase and force push as long as we don't squash the commits. :+1: 

Thanks!

Reply:
>It's ok to rebase and force push as long as we don't squash the commits.

Thanks for the clarification.

All set. Looks like the coverage check is passing now.